### DISCUSSION TITLE
Important selector does not modify

### DISCUSSION THREAD

**Discussion by kolaente** [tailwindlabs/tailwindcss#13069] [Help]
**What version of Tailwind CSS are you using?**

3.4.1

**What build tool (or framework if it abstracts the build tool) are you using?**

Svelte 4.2.8 / Vite 5.0.8

**What version of Node.js are you using?**

20

**What browser are you using?**

Firefox / Chrome

**What operating system are you using?**

NixOS

**Reproduction URL**

https://play.tailwindcss.com/J68KhTQ5z0

**Describe your issue**

I'm building a widget which gets embedded into other websites. These websites may have other style classes set which then override Tailwind's class names. Like this:

```
.p-4 {
  padding: 4rem !important;
}
```

(Yes, they have set their utility classes with `!important`)

I'm using Tailwind's [important selector](https://tailwindcss.com/docs/configuration#selector-strategy) in my config. This works as long as the main website stylesheet does not use `!important` in their interfering class names. Once they do, their styles are more important than mine. I can't change their styles.

Ideally, I'd love a way to set a selector _and_ apply the `!important` modifier to all styles at the same time. That way, I won't override the website's styles and they won't override mine. I know I can add `!` to all classes, but that's a bit tedious and will be one more thing I need to think about while building this.

Maybe similar to https://github.com/tailwindlabs/tailwindcss/issues/11331

**Comment by adamwathan** [comment]
Hey! At the moment this behavior is design and not something we consider a bug, so going to move it over here to discussions.

I was playing around to see if there was a solution to this in user-land and came up with this:

https://play.tailwindcss.com/Sc0NhDgMoE?file=css

If you did that and ran everything through something that handles nesting for you afterwards (assuming you aren't just targeting super modern browsers that have nesting support built in), you should get the results you want.

Right now there are some warnings that get issues but we should just clean that up now that nesting is a native CSS feature.
